packages/find-root/src/index.ts

```diff
@@ -20,34 +20,56 @@ import {
  * monorepo implementations first, with tools based on custom file schemas
  * checked last.
  */
-const defaultOrder: Tool[] = [
+export const DEFAULT_TOOLS = [
   YarnTool,
   BoltTool,
   PnpmTool,
   LernaTool,
   RushTool,
+  RootTool,
 ];
 
 const isNoEntryError = (err: unknown): boolean =>
   !!err && typeof err === "object" && "code" in err && err.code === "ENOENT";
 
 export class NoPkgJsonFound extends Error {
+  directory: string;
+  constructor(directory: string) {
+    super(`No package.json could be found upwards from directory ${directory}`);
+    this.directory = directory;
+  }
+}
+
+export class NoMatchingMonorepoFound extends Error {
   directory: string;
   constructor(directory: string) {
     super(
-      `No package.json could be found upwards from the directory ${directory}`
+      `No monorepo matching the list of supported monorepos could be found upwards from directory ${directory}`
     );
     this.directory = directory;
   }
 }
 
-export async function findRoot(cwd: string): Promise<MonorepoRoot> {
+/**
+ * Given a starting folder, search that folder and its parents until a supported monorepo
+ * is found, and return a `MonorepoRoot` object with the discovered directory and a
+ * corresponding monorepo `Tool` object.
+ *
+ * By default, all predefined `Tool` implementations are included in the search -- the
+ * caller can provide a list of desired tools to restrict the types of monorepos discovered,
+ * or to provide a custom tool implementation.
+ */
+export async function findRoot(
+  cwd: string,
+  tools?: Tool[]
```

Comment:
I'd probably go with an object argument right from the start:
```suggestion
  { tools = DEFAULT_TOOLS }: { tools?: Tool[] }
```

Reply:
That's fair, I had written it this way just because the signature was identical to `get-packages`, which I thought made it maybe easy to see that they are the same signature.

(In the case of `get-packages`, I do want to pass along the undefined, so the default order isn't locked in from two different code paths.)

Reply:
Yeah, totally - `get-packages` should use the default param for this. Otherwise those signatures could still be "the same"